libbeat/publisher/queue/diskqueue/reader_loop.go

```diff
@@ -71,13 +70,15 @@ type readerLoop struct {
 }
 
 func newReaderLoop(settings Settings) *readerLoop {
+	decoder := newEventDecoder()
+	decoder.SetCompression(settings.UseCompression)
```

Comment:
The compression field can't be set based on the global queue settings -- the reader might be processing a segment created by a previous run of the program, or even a previous version, so the compression flag needs to be set for each segment from the metadata in the segment header.

Reply:
> The compression field can't be set based on the global queue settings -- the reader might be processing a segment created by a previous run of the program, or even a previous version, so the compression flag needs to be set for each segment from the metadata in the segment header.

Absolutely.  Good catch.  For right now, it lets me test compression/no compression.  But yeah if it ends up being something configurable we need to go with either a new schema version or header option